src/modules/paymentProcessor/PP_Queue_v1.sol

```diff
@@ -0,0 +1,941 @@
+        (bool success, bytes memory data) = order_.order.paymentToken.call(
+            abi.encodeWithSelector(
+                IERC20(order_.order.paymentToken).transferFrom.selector,
+                order_.client,
+                order_.order.recipient,
+                order_.order.amount
+            )
+        );
+
+        // If transfer was successful.
+        if (
+            success && (data.length == 0 || abi.decode(data, (bool)))
+                && order_.order.paymentToken.code.length != 0
+        ) {
+            _updateOrderState(orderId_, RedemptionState.COMPLETED);
+            _removeFromQueue(orderId_);
+
+            // Notify client about successful payment.
+            IERC20PaymentClientBase_v1(order_.client).amountPaid(
+                order_.order.paymentToken, order_.order.amount
+            );
+
+            emit TokensReleased(
+                order_.order.recipient,
+                order_.order.paymentToken,
+                order_.order.amount
+            );
+
+            return true;
+        } else {
+            // Store as unclaimable and update state.
+            _unclaimableAmountsForRecipient[order_.client][order_
+                .order
+                .paymentToken][order_.order.recipient] += order_.order.amount;
+
+            emit UnclaimableAmountAdded(
+                order_.client,
+                order_.order.paymentToken,
+                order_.order.recipient,
+                order_.order.amount
+            );
+
+            _updateOrderState(orderId_, RedemptionState.CANCELLED);
+            _removeFromQueue(orderId_);
+
+            return false;
+        }
+    }
+
+    /// @notice	Executes all pending orders in the queue.
+    function _executePaymentQueue(address client_) internal {
+        _ensureValidClient(client_);
+        uint firstId = _queue[client_].getNextId(LinkedIdList._SENTINEL);
+        if (firstId == LinkedIdList._SENTINEL) {
+            revert Module__PP_Queue_EmptyQueue();
+        }
+
+        uint processedCount;
+        while (firstId != LinkedIdList._SENTINEL && _processNextOrder(client_))
+        {
+            ++processedCount;
+            firstId = _queue[client_].getNextId(LinkedIdList._SENTINEL);
+        }
+
+        emit PaymentQueueExecuted(_msgSender(), client_, processedCount);
+    }
+
+    /// @notice	Adds a payment order to the queue.
+    /// @param	order_ The payment order to add.
+    /// @param  client_ The client paying for the order.
+    /// @return	orderId_ The ID of the added order.
+    function _addPaymentOrderToQueue(
+        IERC20PaymentClientBase_v1.PaymentOrder memory order_,
+        address client_
+    ) internal returns (uint orderId_) {
+        _ensureValidClient(client_);
+        // Check queue size
+        if (_queue[client_].length() >= _maxQueueSize) {
+            revert Module__PP_Queue_QueueOperationFailed(client_);
+        }
+
+        // Create new order
+        orderId_ = _nextOrderId++;
```

Comment:
Should make use of the hashed OrderId

Reply:
I've implemented it in the branch which I will push in a sec. However I'm now seeing/questioning if it would work, given the LinkedList needs a `uint256` for the order ID

Reply:
We move forward with a orderID which is an `uint256`, so no need for hashing